hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/web/oauth2/ConfRefreshTokenBasedAccessTokenProvider.java

```diff
@@ -103,36 +104,39 @@ public synchronized String getAccessToken() throws IOException {
 
   void refresh() throws IOException {
     try {
-      OkHttpClient client = new OkHttpClient();
-      client.setConnectTimeout(URLConnectionFactory.DEFAULT_SOCKET_TIMEOUT,
-          TimeUnit.MILLISECONDS);
-      client.setReadTimeout(URLConnectionFactory.DEFAULT_SOCKET_TIMEOUT,
-                TimeUnit.MILLISECONDS);
+      OkHttpClient client =
+          new OkHttpClient.Builder().connectTimeout(URLConnectionFactory.DEFAULT_SOCKET_TIMEOUT,
+                  TimeUnit.MILLISECONDS)
+              .readTimeout(URLConnectionFactory.DEFAULT_SOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
+              .build();
```

Comment:
Would you use try-with-resources for `OkHttpClient`?

Reply:
Added here

`try (Response responseBody = client.newCall(request).execute()) {`

Reply:
I meant starting with
```
  try (OkHttpClient client =
          new OkHttpClient.Builder().connectTimeout(URLConnectionFactory.DEFAULT_SOCKET_TIMEOUT,
                  TimeUnit.MILLISECONDS)
              .readTimeout(URLConnectionFactory.DEFAULT_SOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
              .build()) {
```

Reply:
try-with-resources  need Required type: AutoCloseable, so adding the following is giving error

```
  try (OkHttpClient client =
          new OkHttpClient.Builder().connectTimeout(URLConnectionFactory.DEFAULT_SOCKET_TIMEOUT,
                  TimeUnit.MILLISECONDS)
              .readTimeout(URLConnectionFactory.DEFAULT_SOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
              .build()) {
```

and that's why I added on Added here

```
try (Response responseBody = client.newCall(request).execute()) {
```

Ref: https://square.github.io/okhttp/recipes/

Reply:
Cool. Sorry my comment was wrong.